### DISCUSSION TITLE
`ul` styled with `disc inside` irregular indentation

### DISCUSSION THREAD

**Discussion by jonohewitt** [mdx-js/mdx#2095] [Q&A]
Hi everyone, in my MDX code I have a `ul` with two `li` items, each with a component but one followed by additional plain text e.g:
```jsx
  - <span>Example</span>
  - <span>Example</span> .
```

When I style this `ul` with `list-style: disc inside`, I somehow get additional `whitespace`  added before the start of the component text only on the `li` that is not followed by plain text. This leads to choppy indentation, e.g

<img width="129" alt="Screenshot 2022-08-06 at 23 06 38" src="https://user-images.githubusercontent.com/47791294/183266024-4c5fcf79-7ca1-4f74-adf8-ad3ded4faa58.png">

<img width="227" alt="Screenshot 2022-08-06 at 23 04 46" src="https://user-images.githubusercontent.com/47791294/183265980-edd88bdc-e898-4b90-aefd-bc40781b4519.png">

Does anyone have any ideas of what might be causing this and what I could do to fix it? Thank you!

**Comment by wooorm** [comment]
This is, at the root, likely due to how interleaving works. https://mdxjs.com/docs/what-is-mdx/#interleaving. In essence, “same line or not”. That makes the first item a “block” on its own. And the second item a “paragraph” that contains the components and the `.`. Then, list items in markdown come into play. Because there are no blank lines, these are tight list items. Meaning that there the paragraphs are not output. At some point, I’d have to dig into things, those line endings are inserted, but differently due to “paragraph or not”.

If you don’t like the interleaving, you can always force JSX by opening expressions, e.g.,

```jsx
- {<><span>Example</span></>}
- {<><span>Example</span>.</>}
```

…or the whole list in JSX maybe?
